### DISCUSSION TITLE
ValueError: Excel file format cannot be determined, you must specify an engine manually.

### DISCUSSION THREAD

**Discussion by Constable1819** [pypsa-meets-earth/pypsa-earth#1099] [Q&A] [answered]
I have been struggling with this error for weeks now. Please help.

ValueError: Excel file format cannot be determined, you must specify an engine manually.

**Comment by ekatef** [comment, marked as answer]
Hello @Constable1819 and many thanks for the report. 

I assume you get the error in `build_powerplants`, correct? If that is the case the problem are caused by some troubles with an assess to one of the powerplants databases. Currently, the problem is fixed in `powerplantsmatching` but this fix has not yet been included into release which means that some additional tricks are needed to make it work. 

@swaechchha has tested the following procedure for that:
1) comment out [powerplantmatching>=0.5.7](https://github.com/pypsa-meets-earth/pypsa-earth/blob/2e2fdc289932b386f900265c0fc1283518d3fa48/envs/environment.yaml#L18) line from `environment.yaml` specification
2) add the following line to install `powerplantmatching` from master branch of its github repo
```python
- pip:
  - git+https://github.com/pypsa/powerplantmatching@master
  ```
3) remove cache `~/.local/share/powerplantmatching/`
4) build a new environment (best to use a modified environment name to keep a previous version of the environment).

As a quick fix, you can comment out in [configs/powerplantmatching_config.yaml](https://github.com/pypsa-meets-earth/pypsa-earth/tree/main/configs) the datasets of Global Energy Monitoring, namely
 GBPT, GGPT, GCPT, GGTPT, GNPT, GSPT, GWPT, GHPT. However, that will likely lead to a significant data loss and may be recommended only as a temporally solution.

**Reply by Constable1819** [reply]
Hi... This worked, thank you

**Comment by FabianHofmann** [comment]
The problem comes from GEM and their urls being more restrictive (happened last week THU). We will release a new powerplantmaching version today which will resolve this
